Assets/LeapMotion/Scripts/Attachments/AttachmentHand.cs

```diff
@@ -0,0 +1,429 @@
+                                                     middleDistalJoint,
+                                                     middleMiddleJoint,
+                                                     middleTip);
+      if (topLevelTransform != null) {
+        topLevelTransform.SetSiblingIndex(siblingIdx++);
+      }
+
+      // Ring
+      topLevelTransform = tryStackTransformHierarchy(ringKnuckle,
+                                                     ringDistalJoint,
+                                                     ringMiddleJoint,
+                                                     ringTip);
+      if (topLevelTransform != null) {
+        topLevelTransform.SetSiblingIndex(siblingIdx++);
+      }
+
+      // Pinky
+      topLevelTransform = tryStackTransformHierarchy(pinkyKnuckle,
+                                                     pinkyDistalJoint,
+                                                     pinkyMiddleJoint,
+                                                     pinkyTip);
+      if (topLevelTransform != null) {
+        topLevelTransform.SetSiblingIndex(siblingIdx++);
+      }
+    }
+
+    private static Transform[] s_hierarchyTransformsBuffer = new Transform[4];
+    /// <summary>
+    /// Tries to build a parent-child stack (index 0 is the first parent) of the argument
+    /// transforms (they might be null) and returns the number of transforms that were non-null.
+    /// </summary>
+    private Transform tryStackTransformHierarchy(params Transform[] transforms) {
+      for (int i = 0; i < s_hierarchyTransformsBuffer.Length; i++) {
+        s_hierarchyTransformsBuffer[i] = null;
+      }
+
+      int hierarchyCount = 0;
+
+      foreach (var transform in transforms.Query().Where(t => t != null)) {
+        s_hierarchyTransformsBuffer[hierarchyCount++] = transform;
+      }
+
+      for (int i = hierarchyCount - 1; i > 0; i--) {
+        s_hierarchyTransformsBuffer[i].parent = s_hierarchyTransformsBuffer[i - 1];
+      }
+
+      if (hierarchyCount > 0) {
+        return s_hierarchyTransformsBuffer[0];
+      }
+
+      return null;
+    }
+
+    private static Transform[] s_transformsBuffer = new Transform[4];
+    private Transform tryStackTransformHierarchy(params MonoBehaviour[] monoBehaviours) {
+      for (int i = 0; i < s_transformsBuffer.Length; i++) {
+        s_transformsBuffer[i] = null;
+      }
+
+      int tIdx = 0;
+      foreach (var behaviour in monoBehaviours.Query().Where(b => b != null)) {
+        s_transformsBuffer[tIdx++] = behaviour.transform;
+      }
+
+      return tryStackTransformHierarchy(s_transformsBuffer);
+    }
+
+    [ThreadStatic]
+    private static AttachmentPointBehaviour[] s_attachmentPointsBuffer = new AttachmentPointBehaviour[32];
+
+    /// <summary>
+    /// An enumerator that traverses all of the existing AttachmentPointBehaviours beneath an
+    /// AttachmentHand.
+    /// </summary>
+    public struct AttachmentPointsEnumerator {
```

Comment:
Because you are using a static backing array it makes it impossible to have two enumerators active at the same time.